x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/WaterfallItem.tsx

```diff
@@ -102,16 +96,61 @@ interface IWaterfallItemProps {
   onClick: () => any;
 }
 
-function Prefix({ item }: { item: IWaterfallItem }) {
-  if (item.docType !== 'transaction') {
+function PrefixIcon({ item }: { item: IWaterfallItem }) {
+  if (item.docType === 'span') {
+    // icon for database spans
+    const isDbType = item.span.span.type === 'db';
```

Comment:
I noticed the database icon isn't showing up, so I dug in here. For one of the Java transactions, this type value is "db.postgresql.query" at this point. Looks like we either need to duplicate the type conversion being done here: https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/SpanFlyout/StickySpanProperties.tsx#L29-L31, move that conversion higher up so the span.type is always set correctly everywhere, or just check if `type.startsWith("db")` or "db." or something like that.

Reply:
`type.startsWith("db.")` would work too